### ISSUE TITLE
Formatter next steps

### ISSUE THREAD

**Issue by sornas** [FredTheDino/sylt-lang#374]
The formatter (#344) should now be able to take any valid program and output an equivalent program. However, there are still some things left to do with the output.

- [ ] Rewrite to use moves instead of borrows: https://github.com/FredTheDino/sylt-lang/pull/344#discussion_r699564507.
- [ ] Comment placement.
- [ ] Empty blob declarations (`c := C {}`).
- [ ] EmptyStatement becomes blank lines where they shouldn't. (Mostly as the first statement in a block since we allow `{ f() }`.)
- [ ] `f()` can be written `f'`, usually (but not always).
- [ ] `int | nil| can be written `int?`.
- [ ] Blank lines separating use-statements, functions etc.
- [ ] Short lists inline, longer lists with one element per line. (`[1,2,3]` doesn't need multiple lines.)
- [ ] Implicit returns.

**Comment by FredTheDino** [comment]
Are trailing whitespace still a problem?

**Comment by sornas** [comment]
> Are trailing whitespace still a problem?

Yes, but only for EmptyStatement. Technically a subset of "EmptyStatement becomes blank lines where they shouldn't."
